### PR TITLE
204a unicode filepath support

### LINKED ISSUE #204: Add support for installing bundles from non-ascii paths on Windows.
bundle locations with non-ASCII characters are won't install into the framework. As long as the path provided to the framework is a valid UTF-8 string, the framework should be able to install and start the bundle without any issues.

### PR DESCRIPTION
**Pull request by karthikreddy09**
fixes #204

This PR is a continuation of https://github.com/CppMicroServices/CppMicroServices/pull/207
The original PR will be discarded once this one is merged in.

### CODE REVIEW

File: third_party/miniz.c [hunk 1]
```diff
@@ -2850,17 +2850,52 @@ void *tdefl_write_image_to_png_file_in_memory(const void *pImage, int w, int h,
   #include <sys/stat.h>
 
   #if defined(_MSC_VER) || defined(__MINGW64__)
+    #include <Windows.h>
+    #include <stringapiset.h>
+    #include <wchar.h>
+    
+    // return NULL if inStr is NULL or if the conversion failed
```
Comment (jeffdiclemente) [inline third_party/miniz.c]: a note about who is responsible for de-allocating the memory returned by this function is important.

File: util/src/String.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#include <cppmicroservices/GlobalConfig.h>
```
Comment (ksubramz) [inline util/src/String.cpp]: Missing Copyright and License header

File: framework/test/driver/CMakeLists.txt [hunk 3]
```diff
@@ -152,3 +152,12 @@ if(US_BUILD_SHARED_LIBS)
     add_dependencies(${_test_driver} ${_test_bundle})
   endforeach()
 endif()
+
+
+#-----------------------------------------------------------------------------
+# Add preprocessor macro to indicate C++11 unicode literal support
+#-----------------------------------------------------------------------------
+
+target_compile_definitions(${_test_driver} PUBLIC
```
[the remark this answers is not in the document]
Comment (karthikreddy09) [reply framework/test/driver/CMakeLists.txt]: It's not supported in VS2013. Whats the value of adding the check in a top-level file as opposed to here?

[the remark this answers is not in the document]
Comment (karthikreddy09) [reply framework/test/driver/CMakeLists.txt]: had an offline conversation with Jeff to clarify what is done here.

[the remark this answers is not in the document]
Comment (jeffdiclemente) [reply framework/test/driver/CMakeLists.txt]: there's no value. I misunderstood how unicode literal support detection worked here. This looks fine to me.

**Pull request merged**